### PR TITLE
[nasa/nos3#597] Sample system test and automated scenario tests (ASTs)

### LINKED ISSUE #597: Sample Instrument System Tests Update
### Summary

System Test to check operation and proper telemetry responses to all commands in COSMOS.

### Use Case

Allows us to check operation and proper telemetry responses to all commands in COSMOS.

### PR DESCRIPTION
**Pull request by jlucas9**
Closes #597 .

What does it do?
* Updated system tests to use a library and confirm additional edge cases
* Migrated SIM_CMDBUS_BRIDGE file into component
* Created automated scenario tests (ASTs) that utilize the simulator command bus bridge to inject faults
* Change default GSW back to COSMOS while ASTs are realized across subsystems

How to test?
* Checkout branch and `git submodule update --init --recursive`
* `make clean; make; make launch`
* Start up COSMOS Test Runner
* Change to "Test Suite: Sample_Test" and click "Start" at that Test Suite level
* Confirm all tests pass
* Repeat as desired

image

Any submodule PRs and actions prior to closing this?
* https://github.com/nasa-itc/sample/pull/28
* https://github.com/nasa-itc/gsw-cosmos/pull/25

### CODE REVIEW

**Review by Isgaroth-the-Green — APPROVED** [review]
All the tests worked fine for me, and everything looks good.

**Comment by zlynch2** [comment]
Are we looking to get cmdbus_bridge working for all the other components here? @jlucas9 meaning when we translate this format to other components system tests.

**Comment by dacarter22** [comment]
image
Copied sample files into css and trying to rebuild tests around them now but I ran into this error and I think the only way to fix it is to change how commands are being counted in the css.app. My question is, should we change the standards for how commands are counted (double disable not counting in the new sample.app) or should we change the testing standards to fit the current apps in the other components

**Comment by jlucas9** [comment]
> Are we looking to get cmdbus_bridge working for all the other components here? @jlucas9 meaning when we translate this format to other components system tests.

Yes please, I think that cmdbus_bridge would be needed for everything to get a real full system test suite.

**Comment by jlucas9** [comment]
> image Copied sample files into css and trying to rebuild tests around them now but I ran into this error and I think the only way to fix it is to change how commands are being counted in the css.app. My question is, should we change the standards for how commands are counted (double disable not counting in the new sample.app) or should we change the testing standards to fit the current apps in the other components

Go ahead and resolve if you can. Ideally no double counts would be occurring and all components comply to the same standard.

**Review by dacarter22 — APPROVED** [review]

File: scripts/gsw/gsw_cosmos_launch.sh [hunk 1]
```diff
@@ -19,4 +19,4 @@ export GSW="cosmos_openc3-operator_1"
 #mkdir /tmp/uplink 2> /dev/null
 
 echo "COSMOS launch..."
-gnome-terminal --tab --title="Cosmos" -- $DFLAGS -v $BASE_DIR:$BASE_DIR -v /tmp/nos3:/tmp/nos3 -v /tmp/.X11-unix:/tmp/.X11-unix:ro -e DISPLAY=$DISPLAY -e QT_X11_NO_MITSHM=1 -w $GSW_DIR --name cosmos_openc3-operator_1 --network=nos3_core ballaerospace/cosmos:4.5.0
+gnome-terminal --tab --title="Cosmos" -- $DFLAGS -v $BASE_DIR:$BASE_DIR -v /tmp/nos3:/tmp/nos3 -v /tmp/.X11-unix:/tmp/.X11-unix:ro -e DISPLAY=$DISPLAY -e QT_X11_NO_MITSHM=1 -e PROCESSOR_ENDIANNESS="LITTLE_ENDIAN" -w $GSW_DIR --name cosmos_openc3-operator_1 --network=nos3_core ballaerospace/cosmos:4.5.0
```
Comment (jlucas9) [inline scripts/gsw/gsw_cosmos_launch.sh:22]: Possibly rename to flight processor endianness

**Review by kevincbruce — APPROVED** [review]

**Review by zlynch2 — APPROVED** [review]
per code review today and testing things look good, saw john added things from code review today. Approving

**Review by dacarter22 — APPROVED** [review]
Watched code added during meeting, re-pulled, re-ran, and everything works, even with my submodule branches, approving
